pyrobosim/pyrobosim/planning/pddlstream/utils.py

```diff
@@ -156,6 +157,11 @@ def pddlstream_solution_to_plan(solution, robot):
                 arg = act_pddl.args[3]
                 if isinstance(arg, Pose):
                     act.pose = arg
+            # If a grasp pose is specified, add that instead
+            if act.type == "pick" and len(act_pddl.args) > 5:
+                arg = act_pddl.args[5]
+                if isinstance(arg, Grasp):
+                    act.pose = arg.origin
```

Comment:
I deeply dislike these rules. If you have any thoughts on how to better convert an arbitrary PDDLStream plan to an actual action plan, I'd be interested in hearing it.

Even if we move the PDDLStream mappings/primitives to a user workspace, users will also need to touch this code if they're doing anything "non-standard" per our examples.

Reply:
probably functional programming would help, i.e. the user can pass a list of functions to `pddlstream_solution_to_plan()`. Most of those functions will probably come from the pyrobosim module but sometimes it might come from the user's custom module.

I'm not an expert on that at all, especially not in Python

Reply:
That's very good! You can just pass a dictionary mapping from the action name to a function that makes the action itself.

In fact, that's exactly how PDDLStream does the mappings, as seen in the `mappings.py` file